callbacks/create.go

```diff
@@ -103,9 +103,15 @@ func Create(config *Config) func(db *gorm.DB) {
 		}
 
 		db.RowsAffected, _ = result.RowsAffected()
-		if db.RowsAffected != 0 && db.Statement.Schema != nil &&
-			db.Statement.Schema.PrioritizedPrimaryField != nil &&
-			db.Statement.Schema.PrioritizedPrimaryField.HasDefaultValue {
+		if db.RowsAffected == 0 {
+			return
+		}
+
+		if db.Statement.Schema != nil {
+			if db.Statement.Schema.PrioritizedPrimaryField == nil || !db.Statement.Schema.PrioritizedPrimaryField.HasDefaultValue {
+				return
+			}
+
 			insertID, err := result.LastInsertId()
```

Comment:
In some cases,  the `Dest` might be a map value, but `db.Statement.Schema` is not nil, for example:


`db.Model(&User{}).First(map[string]interface{}{})`

Reply:
btw, can you add some test cases for the feature, thank you.